packages/mui-material/src/SwipeableDrawer/SwipeableDrawer.js

```diff
@@ -227,7 +227,8 @@ const SwipeableDrawer = React.forwardRef(function SwipeableDrawer(inProps, ref)
   );
 
   const handleBodyTouchEnd = useEventCallback((nativeEvent) => {
-    if (!touchDetected.current) {
+    // the ref may be null when a parent component updates while swiping
+    if (!paperRef.current || !touchDetected.current) {
```

Comment:
This type of defensive check usually hides root causes. Here, I suspect that because the cleanup happens in a useeffect, it's async, will it should be a layout effect

https://github.com/mui-org/material-ui/blob/fcf2a8a5c90e6a975b0a8c391aa2f95f6e860b15/packages/mui-material/src/SwipeableDrawer/SwipeableDrawer.js#L510

Reply:
Hey @oliviertassinari, thanks for you answer, I definitely agree. We've inlined the component in production to check, and it's not the root cause anyway, the sentries are still there, which sounds a bit surprising 🤔  I've also tried to replace this `useEffect` by a `useLayoutEffect, and still the same errors

![image](https://user-images.githubusercontent.com/17592779/147764124-11beabd1-53f0-4436-9c4e-12d17aad065d.png)
![image](https://user-images.githubusercontent.com/17592779/147764462-5b81f783-947f-4470-ba23-c16f43848eef.png)



I've tries various different stuff to trigger the error, but no luck I didn't reproduce it yet. But it's not that edge case as we have 1k/day occurences on iOS and 1k/day on Android. Thus it doesn't look like OS-related, i'd say it's more about the react lifecycle itself